### PR TITLE
using nn.Dropout() layer

### PR DESCRIPTION
**Pull request by mberaha**

### CODE REVIEW

File: resources/local_params.asciipb [hunk 1]
```diff
@@ -9,11 +9,11 @@ temperature: 0.1
 lambda_GAN: 1
 dropout: 0.5
 max_loss: 1e10
-grad_clip: 30
+grad_clip: 20
```
Comment (lorelupo) [inline resources/local_params.asciipb:12]: avevo messo io 30 perchè lo mette anche la zia

File: src/rnn.py [hunk 2]
```diff
@@ -15,12 +17,15 @@ def __init__(self, input_size, hidden_size, num_layers, batch_first):
...
         # create the RNN cell
         self.batch_first = batch_first
+        self.dropoutLayer = nn.Dropout(p=dropout)
         self.cell = nn.GRU(
             self.input_size, self.hidden_size,
             self.num_layers, batch_first=self.batch_first)
         # print("Batch first: ", self.cell.batch_first)
 
     def forward(self, inputs, hidden, lengths=[], pad=True):
+        inputs = self.dropoutLayer(inputs)
```
Comment (lorelupo) [inline src/rnn.py:27]: da valutare post training se questo dropout è di troppo, perchè tutti gli rnn ricevono in input delle frasi a cui è già stato applicato un dropout=0.1 alle parole, che vengono sostituite da <unk> (funzione noise).

**Review by lorelupo — APPROVED** [review]

File: src/rnn.py (hunk 2 above)
Comment (mberaha) [reply src/rnn.py:27]: lei usa questo wrapper:
https://www.tensorflow.org/api_docs/python/tf/contrib/rnn/DropoutWrapper

dalle doc: "Operator adding dropout to inputs and outputs of the given cell."

però ok!

File: resources/local_params.asciipb (hunk 1 above)
Comment (mberaha) [reply resources/local_params.asciipb:12]: colpa mia avevo mega merge issues ed è scappato
